reactive/kotlinx-coroutines-rx2/src/RxConvert.kt

```diff
@@ -81,7 +81,12 @@ public fun <T: Any> ObservableSource<T>.asFlow(): Flow<T> = callbackFlow {
     val observer = object : Observer<T> {
         override fun onComplete() { close() }
         override fun onSubscribe(d: Disposable) { if (!disposableRef.compareAndSet(null, d)) d.dispose() }
-        override fun onNext(t: T) { sendBlocking(t) }
+        override fun onNext(t: T) {
+            try {
+                sendBlocking(t)
+            } catch (ignored: Exception) { //TODO: Replace when this issue is fixed: https://github.com/Kotlin/kotlinx.coroutines/issues/974
```

Comment:
Do we want to catch all `Exceptions` or should we be more specific, e.g., `CancellationException`, `ClosedSendChannelException`?

Reply:
I'm following the example in the KDoc of `callbackFlow`. It's unlikely (impossible?) this call will throw anything else anyway.

Reply:
It can throw `Throwable` or `Error` if the channel was closed with it as a cause. It should generally catch `Throwable`

Reply:
Ah yes, missed the `close(e)` call that has the `Throwable` type.